alpha-build-machinery/make/targets/openshift/crd-schema-gen.mk

```diff
@@ -0,0 +1,24 @@
+CRD_SCHEMA_GEN_TEMP ?=$(shell mktemp -d)
+
+# $1 - schema gen version
+# $2 - manifests directory
+# $3 - package APIs
+define codegen-crds
+crd-schema-gen:
+	mkdir -p $(CRD_SCHEMA_GEN_TEMP)/bin
+	git clone -b $(1) --single-branch --depth 1 https://github.com/kubernetes-sigs/controller-tools.git $(CRD_SCHEMA_GEN_TEMP)/gen
+	cd $(CRD_SCHEMA_GEN_TEMP)/gen; GO111MODULE=on go build ./cmd/controller-gen; mv controller-gen ../bin
+	curl -f -L -o $(CRD_SCHEMA_GEN_TEMP)/bin/yq https://github.com/mikefarah/yq/releases/download/2.4.0/yq_$(shell uname -s | tr A-Z a-z)_amd64 && chmod +x $(CRD_SCHEMA_GEN_TEMP)/bin/yq
+
+update-codegen-crds: crd-schema-gen
+	$(CRD_SCHEMA_GEN_TEMP)/bin/controller-gen \
+		schemapatch:manifests=$(2) \
+		output:dir=$(2) \
+		paths="$(subst $() $(),;,$(3))"
+	for p in manifests/*.crd.yaml-merge-patch; do $(CRD_SCHEMA_GEN_TEMP)/bin/yq m -i "$${p%%.crd.yaml-merge-patch}.crd.yaml" "$$p"; done
+
+verify-codegen-crds: update-codegen-crds
+	git diff -q manifests/ || { echo "Changed manifests: "; echo; git diff; false; }
+
+.PHONY: update-codegen-crds verify-codegen-crds crd-schema-gen
```

Comment:
by convention, in openshift, we use a separate phony statement marking the end of block